packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -62,6 +62,18 @@ export class MigrationDataSource {
     return this.ctx.coreData.migration.legacyConfigForMigration
   }
 
+  get shouldMigratePreExtension () {
+    return !this.legacyConfig.projectId
```

Comment:
I wonder if anyone does nested projectId, like

```json
{
  "e2e": {
    "projectId": "ABC"
  },
  "component": {
    "projectId": "DEF"
  }
}
```
Not sure if this is even valid, but if it is, we might need to consider this scenario, too.

Reply:
According to Brian, this should not be permitted, it needs to be a root level property only

Reply:
We might need to verify if this actually works in 9.x and see if anyone was using it like that, seems unlikely but you never know.

Eg: I'm saying if it *was* previously permitted, it's possible someone thought to do this.

Reply:
I tried the projectId inside the e2e and it works fine in 9.x

The support for this should be something like ` return !this.legacyConfig.projectId && !this.legacyConfig.e2e?.projectId` - how common may this be? I guess it would be safer to support it

Reply:
That means `component.projectId` probably works, too. If someone either `e2e.projectId` OR `component.projectId`, I think we should not rename, just to be safe.

I have no idea how common this is, but since this is how we make $$$ I don't think we can just assume it's unlikely to happen - we need to get this 100% perfect. I think we should do as you suggested, and **also** add a check for `component.projectId` to be 100% safe. Thoughts?

Reply:
For the moment we are not gonna take into account the CT for the history on the cloud - just the E2E tests; that's why I only update the `e2e.projectId`